### ISSUE TITLE
Unexpected keyword argument 'affinity'

### ISSUE THREAD

**Issue by kokyriakidis** [zengxiaofei/HapHiC#23]
I get the following error when I try to run haphic:

```
2024-04-27 13:16:53 <HapHiC_reassign.py> [run] A pickle file is input meanwhile --remove_allelic_links is set, allelic Hi-C links will NOT be treated specially in the reassignment step
2024-04-27 13:16:53 <HapHiC_reassign.py> [parse_pickle] Parsing input pickle file...
2024-04-27 13:16:53 <HapHiC_reassign.py> [parse_clusters] Parsing .clusters.txt file...
2024-04-27 13:16:53 <HapHiC_reassign.py> [run] File parsing and data preparation finished in 144.6104302406311s
2024-04-27 13:16:53 <HapHiC_reassign.py> [run_reassignment] Performing reassignment...
2024-04-27 13:16:53 <HapHiC_reassign.py> [run_reassignment] [result::round1] Total: 21953, consistent: 326, rescued: 237, reassigned: 13, not rescued: 21377
2024-04-27 13:16:53 <HapHiC_reassign.py> [run_reassignment] Performing reassignment...
2024-04-27 13:16:54 <HapHiC_reassign.py> [run_reassignment] [result::round2] Total: 21953, consistent: 538, rescued: 29, reassigned: 2, not rescued: 21384
2024-04-27 13:16:54 <HapHiC_reassign.py> [run_reassignment] Performing reassignment...
2024-04-27 13:16:54 <HapHiC_reassign.py> [run_reassignment] [result::round3] Total: 21953, consistent: 563, rescued: 0, reassigned: 0, not rescued: 21390
2024-04-27 13:16:54 <HapHiC_reassign.py> [run] [result::round3] Result has converged after 2 rounds of reassignment, break
2024-04-27 13:16:54 <HapHiC_reassign.py> [run_reassignment] Performing additional round of rescue...
2024-04-27 13:16:54 <HapHiC_reassign.py> [run_reassignment] [result::additional_rescue] Total: 21953, consistent: 653, rescued: 192, reassigned: 0, not rescued: 21108
2024-04-27 13:16:54 <HapHiC_reassign.py> [run] 3 round(s) of reassignment finished in 0.277385950088501s, average 0.09246198336283366s per round
2024-04-27 13:16:54 <HapHiC_reassign.py> [agglomerative_hierarchical_clustering] Performing additional agglomerative hierarchical clustering...
Traceback (most recent call last):
  File "HapHiC_pipeline.py", line 517, in <module>
    main()
  File "HapHiC/scripts/HapHiC_pipeline.py", line 502, in main
    haphic_reassign(args)
  File "HapHiC/scripts/HapHiC_pipeline.py", line 406, in haphic_reassign
    HapHiC_reassign.run(args, log_file=LOG_FILE)
  File "HapHiC/scripts/HapHiC_reassign.py", line 877, in run
    hc_cluster_dict = agglomerative_hierarchical_clustering(
  File "HapHiC/scripts/HapHiC_reassign.py", line 535, in agglomerative_hierarchical_clustering
    clust = AgglomerativeClustering(n_clusters=nclusters, affinity="precomputed", linkage="average", distance_threshold=None)
TypeError: AgglomerativeClustering.__init__() got an unexpected keyword argument 'affinity'
Traceback (most recent call last):
  File "HapHiC/haphic", line 96, in <module>
    subprocess.run(commands, check=True)
  File "/private/home/kkyriaki/micromamba/envs/haphic/lib/python3.10/subprocess.py", line 526, in run
    raise CalledProcessError(retcode, process.args,

```

Probably this line should be changed to `metric="precomputed"` inside the `HapHiC_reassign.py` script:
```
clust = AgglomerativeClustering(n_clusters=nclusters, affinity="precomputed", linkage="average", distance_threshold=None)
```

**Comment by zengxiaofei** [comment]
Hi @kokyriakidis,
The version of your sklearn is too high. Maybe you did not install the dependencies using the conda files we provided. You could check out the versions we have already tested in [environment_py310.yml](https://github.com/zengxiaofei/HapHiC/blob/main/conda_env/environment_py310.yml) or [environment_py311.yml](https://github.com/zengxiaofei/HapHiC/blob/main/conda_env/environment_py311.yml) to solve the problem.
Best,
Xiaofei

**Comment by zengxiaofei** [comment]
Several users have reported this exception when they tried manually installing the environment. I decided to add an if statement to make HapHiC compatible with higher versions of scikit-learn.

**Comment by kokyriakidis** [comment]
I wasn't able to create the environment using these templates. I had to manually add the packages.

**Comment by zengxiaofei** [comment]
This issue has been fixed in the latest commit [c56c648].(https://github.com/zengxiaofei/HapHiC/commit/c56c6487bc3cd8d29cd3ffee91f39082f89609b6)
